### PR TITLE
docstring for data generation function

### LINKED ISSUE #2: Function for generating data for model

### PR DESCRIPTION
**Pull request by fohy24**
Hey this is the docstring for the data generation function. Please take a look!

### CODE REVIEW

**Comment by andyzhangstat** [comment]
Hi Sam, your docstring looks good. But did you get the latest 'main' before pushing your branch? Your branch seems to 1 commit ahead of main, and  2 commits behind main.

File: src/lr_cd/lr_data_generation.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    n : integer
+        The number of data points.
+    theta : float
+        The true theta coefficient for the logistic regression
+    random_seed : integer
+        Random seed to ensure reproducibility.
+
+    Returns
+    -------
+    A dataframe containing the data points (X, y), where X is the randomly generated regressor values, and y is the corresponding binary target.
```
Comment (Jing-19) [inline src/lr_cd/lr_data_generation.py]: Please update the y to random continuous variables.

File: src/lr_cd/lr_data_generation.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    n : integer
+        The number of data points.
+    theta : float
+        The true theta coefficient for the linear regression
+    random_seed : integer
+        Random seed to ensure reproducibility.
+
+    Returns
+    -------
+    An array containing the data points (X, y), where X is the randomly generated regressor values, and y is the corresponding continuous target.
```
Comment (Jing-19) [inline src/lr_cd/lr_data_generation.py]: Please update it to two 1D array

**Comment by Jing-19** [comment]
Looks good to me! thanks!

**Review by Jing-19 — APPROVED** [review]

File: src/lr_cd/lr_data_generation.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    n : integer
+        The number of data points.
+    theta : float
+        The true theta coefficient for the linear regression
+    random_seed : integer
+        Random seed to ensure reproducibility.
+
+    Returns
+    -------
+    An array that returns X, the randomly generated regressor values, and another array containing y, the corresponding continuous target.
```
Comment (andyzhangstat) [inline src/lr_cd/lr_data_generation.py]: there are two items needs to be returned, one is X  and another one is y. Could you return them separately? sth like:
 Returns
    -------
    X : ndarray
        Feature data matrix of shape (n_samples, n_features).
    y : ndarray
        Response data vector of shape (n_samples,).

File: src/lr_cd/lr_data_generation.py [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+# date: 2024-01-10
+
+def generate_data_lr(n, theta, random_seed=123):
+    """Generate a number of data points base on the theta coefficients.
+
+    Parameters
+    ----------
+    n : integer
+        The number of data points.
+    theta : float
```
Comment (andyzhangstat) [inline src/lr_cd/lr_data_generation.py]: The theta here should be a np array which contains the scalar intercept and coefficient weights vector. OR you can use two input arguments: intercept (float) and coeff (array). See the link below for more: https://scikit-learn.org/stable/modules/generated/sklearn.linear_model.LinearRegression.html#sklearn.linear_model.LinearRegression

File: src/lr_cd/lr_data_generation.py [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+        The true theta coefficient for the linear regression
+    random_seed : integer
+        Random seed to ensure reproducibility.
+
+    Returns
```
Comment (andyzhangstat) [inline src/lr_cd/lr_data_generation.py:17]: I think it may be necessary to include the true intercept and coefficients vector as return. Jing may use it for plotting.

**Comment by fohy24** [comment]
> Hi Sam, your docstring looks good. But did you get the latest 'main' before pushing your branch? Your branch seems to 1 commit ahead of main, and 2 commits behind main.

Merged with main branch just now. Should be good.

**Review by andyzhangstat — APPROVED** [review]

**Review by andyzhangstat — CHANGES_REQUESTED** [review]
The input of function may contain the n, p(in our example is default 2) and seed, we don't need to specify the weights. The output of function may contain the X, y, and true intercept and true coefficient vector.

File: src/lr_cd/lr_data_generation.py (hunk 3 above)
Comment (Jing-19) [reply src/lr_cd/lr_data_generation.py]: I think X should be also one-dimensional. The column of ones should be added inside the fit function. But I agree that the return could be more detailed as what Andy suggests

File: src/lr_cd/lr_data_generation.py (hunk 5 above)
Comment (Jing-19) [reply src/lr_cd/lr_data_generation.py:17]: I only need the intercept and coefficients calculated from the fit function. X and y should be the only two outputs for this function?

**Comment by fohy24** [comment]
> The input of function may contain the n, p(in our example is default 2) and seed, we don't need to specify the weights. The output of function may contain the X, y, and true intercept and true coefficient vector.

Returning the true parameters from a set of randomly generated data requires fitting. It is what the fitting function is for.

File: src/lr_cd/lr_data_generation.py (hunk 3 above)
Comment (fohy24) [reply src/lr_cd/lr_data_generation.py]: fixed

File: src/lr_cd/lr_data_generation.py (hunk 4 above)
Comment (fohy24) [reply src/lr_cd/lr_data_generation.py]: fixed

**Review by andyzhangstat — APPROVED** [review]
